### ISSUE TITLE
[Bug][Steam Deck][Desktop Mode] Lock Screen covers up Steam's virtual keyboard, obstructing its use

### ISSUE THREAD

**Issue by Sijiba** [ValveSoftware/SteamOS#825]
#### Your system information

* Steam client version: 1659369000
* SteamOS version: 3.3
* Opted into Steam client beta?: No
* Opted into SteamOS beta?: No
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:

The problem occurs when using a handheld Steam Deck in desktop mode. The lock screen provided through desktop mode's System Settings hides the Steam virtual keyboard while the system is locked, preventing the user from viewing or touching the keyboard.

This makes unlocking the system in desktop mode needlessly difficult. The keyboard can still be navigated with the controller buttons and trackpads only. With enough effort, it can technically still be navigated to input a password through audio/tactile feedback and memorizing the layout and cursor position, but this isn't practical or sufferable for everyday use.

#### Steps for reproducing this issue:

1. Switch to Desktop Mode on a handheld Steam Deck
2. Open System Settings
3. Navigate to Personalization > Users
4. Set the current account's password to something, such as "11111"
5. Navigate to Workspace > Workspace Behavior > Screen Locking
6. Check "After waking from sleep" under "Lock screen automatically:" and apply changes. 
7. Press the power button once, let the screen turn off, then press it again to show the lock screen.
8. Press "Steam"+"X" to activate the virtual keyboard.
Virtual keyboard takes input from the user, but isn't visible or touchable on the screen.

**Comment by syndr** [comment]
A workaround for this is to install the package for qtVirtualKeyboard, which has built-in support with the SDDM display manager (which handles the lock screen).  This will put a "Virtual Keyboard" button in the bottom left of the lock screen.

You can use this script to do so:
```
#!/bin/bash

# This does initial customization for Steam Deck for initial setup / post update config

echo "Setting file system to writable"
sudo steamos-readonly disable

echo "Initializing Pacman"
sudo pacman-key --init
sudo pacman-key --populate
sudo pacman -Syu

echo "Installing on-screen keyboard"
sudo pacman -S qt5-virtualkeyboard --noconfirm

echo "Setting file system back to read-only"
sudo steamos-readonly enable
```

You'll need to have a password set for your user first, which you can do using the KDE "System Settings" GUI app, or the `passwd` command in the shell.

Also note that since this modifies system files you'll need to re-run the script after any major system updates as these changes to the system partition will be lost during the update.

**Comment by k1lln1n3** [comment]
While there is a workaround I'm wondering if this could be looked at. As it stands if I leave my Steam deck in desktop mode and turn it on when I'm not home. I have to turn it off and on again. Not wanting to get a tiny Bluetooth keyboard but I might have to

**Comment by jmerkle** [comment]
It baffles me that this bug has been open since August and apparently hasn't been picked up yet. 
It's such an egregious security flaw. I either have to leave my steam deck unlocked when it's in desktop mode, or i'm forced to reboot it if i don't have a keyboard with me. The fact that there's not even a reboot button on the lock screen and i have to force reboot it with the power button makes this even more ridiculous.

Having to run above script everytime there's a steamOS update does not seem like an acceptable workaround for this.

**Comment by syndr** [comment]
Yeah, for something as obvious as this, I'd really hope that this gets fixed too. We shouldn't have to depend on any sort of workaround for ability to unlock the device.

I get lock screens not being the first thought on a traditional game console, but that's not the Steam Deck. And game mode has a working one already!

**Comment by Qwerty-Space** [comment]
This really needs to be fixed, or qt5-virtualkeyboard needs to come pre-installed

**Comment by walkertr0n** [comment]
This is absurd. Please fix this, Valve.

**Comment by syndr** [comment]
> This is absurd. Please fix this, Valve.

Hey, hey, go easy on them man. It's only been a year!

(/s)

**Comment by Qwerty-Space** [comment]
I think this is a limitation of the kde lockscreen, not necessarily something Valve can fix.

**Comment by jg00dman** [comment]
Having the same problem.

**Comment by robbarclay** [comment]
Same problem, needs a more permanent solution.

Thank you syndr for the workaround though!

**Comment by robbarclay** [comment]
> I think this is a limitation of the kde lockscreen, not necessarily something Valve can fix.

They can literally include virtual keyboard package and it's solved.

**Comment by HybridZach** [comment]
> A workaround for this is to install the package for qtVirtualKeyboard, which has built-in support with the SDDM display manager (which handles the lock screen). This will put a "Virtual Keyboard" button in the bottom left of the lock screen.

Thanks this works perfectly. However does anyone know how to invoke this keyboard outside of the lock screen? this keyboard is far superior to the steam one within desktop mode since it allows for swipe typing.

**Comment by Maelstromeous** [comment]
Absolutely baffles me how Valve haven't installed this package as part of the core desktop experience. It's a MASSIVE security risk!!

**Comment by xfechx** [comment]
Hello?
2025 and still the same issue?
A lot of people are using Desktop Mode, why has this not been looked at?
